compiler/noirc_frontend/src/elaborator/statements.rs

```diff
@@ -110,12 +113,60 @@ impl<'context> Elaborator<'context> {
         (HirStatement::Let(let_), Type::Unit)
     }
 
-    pub(super) fn elaborate_constrain(&mut self, stmt: ConstrainStatement) -> (HirStatement, Type) {
-        let expr_span = stmt.0.span;
-        let (expr_id, expr_type) = self.elaborate_expression(stmt.0);
+    pub(super) fn elaborate_constrain(
+        &mut self,
+        mut stmt: ConstrainStatement,
+    ) -> (HirStatement, Type) {
+        let span = stmt.span;
+        let min_args_count = match stmt.kind {
+            ConstrainKind::Assert | ConstrainKind::Constrain => 1,
+            ConstrainKind::AssertEq => 2,
+        };
+        let max_args_count = min_args_count + 1;
+        let actual_args_count = stmt.arguments.len();
+
+        let (message, expr) = if actual_args_count < min_args_count
+            || actual_args_count > max_args_count
+        {
+            self.push_err(TypeCheckError::AssertionParameterCountMismatch {
+                expected1: min_args_count,
+                expected2: max_args_count,
+                found: actual_args_count,
+                span,
+            });
```

Comment:
`expected1` and `expected2` sound like vague names for this field. Maybe just only pass the minimum argument count since the max is always that + 1? Or rename the fields to min/max arg count.